gate-core/src/main/groovy/com/netflix/spinnaker/gate/model/manageddelivery/ConstraintState.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.model.manageddelivery;
+
+import java.time.Instant;
+import java.util.Map;
+import lombok.Data;
+
+@Data
+public class ConstraintState {
+  String deliveryConfigName;
+  String environmentName;
+  String artifactVersion;
+  String type;
+  String status;
+  Instant createdAt;
+  String judgedBy;
+  Instant judgedAt;
+  String comment;
+  Map<String, Object> attributes;
```

Comment:
Did you mean for this to collect any properties that are added later, or just specific things under attributes?

Reply:
It's a generic type that inherits from `ConstraintStateAttributes` in keel.